src/app/interceptors/auth.interceptor.ts

```diff
@@ -0,0 +1,22 @@
+import type { HttpEvent, HttpHandler, HttpInterceptor, HttpInterceptorFn, HttpRequest } from '@angular/common/http';
+import { Injectable, inject } from '@angular/core';
+import { AuthService } from '../auth/auth.service';
+
+export const authInterceptor: HttpInterceptorFn = (req, next) => {
+  if(req.headers.get('skip')){
+    return next(req);
+  }
+
+  const token = inject(AuthService).getToken();
+
+  if(token){
```

Comment:
There you should check if token exists and it is not expired, than clone the request, if it is expired, than refresh the token